core/src/subject/SubjectEntity.ts

```diff
@@ -0,0 +1,250 @@
+  author: string;
+  timestamp: string;
+
+  constructor(perspective: PerspectiveProxy, baseExpression?: string) {
+    this.#baseExpression = baseExpression ? baseExpression : `entry://${makeRandomPrologAtom(24)}`;
+    this.#perspective = perspective
+  }
+
+  get baseExpression() {
+    return this.#baseExpression
+  }
+
+  private async getData(id?: string) {
+    const tempId = id ?? this.#baseExpression;
+    let isInstance = await this.#perspective.isSubjectInstance(tempId, this.#subjectClass)
+    if (!isInstance) {
+      throw `Not a valid subject instance of ${this.#subjectClass} for ${tempId}`
+    }
+
+    const subject = await this.#perspective.getSubjectProxy(tempId, this.#subjectClass);
+
+    if (!subject) {
+      await this.#perspective.createSubject(this.#subjectClass, tempId);
+    }
+
+    let results = await this.#perspective.infer(`subject_class("${this.#subjectClass}", C), property(C, Property)`)
+    let properties = results.map(result => result.Property)
+
+    for (let p of properties) {
+      const resolveExpressionURI = await this.#perspective.infer(`subject_class("${this.#subjectClass}", C), property_resolve(C, "${p}")`)
+      const getProperty = async () => {
+        let results = await this.#perspective.infer(`subject_class("${this.#subjectClass}", C), property_getter(C, "${tempId}", "${p}", Value)`)
+        if (results && results.length > 0) {
+          let expressionURI = results[0].Value
+
+          if (resolveExpressionURI) {
+            const expression = await this.#perspective.getExpression(expressionURI)
+            this['author'] = expression.author;
+            this['timestamp'] = expression.timestamp;
+            try {
+              return JSON.parse(expression.data)
+            } catch (e) {
+              return expression.data
+            }
+          } else {
+            return expressionURI
+          }
+        } else if (results != undefined) {
+          if (results) {
+            return results
+          } else {
+            return false
+          }
+        } else {
+          return undefined
+        }
+      };
+
+      this[p] = await getProperty()
+    }
+
+    let results2 = await this.#perspective.infer(`subject_class("${this.#subjectClass}", C), collection(C, Collection)`)
+    if (!results2) results2 = []
+    let collections = results2.map(result => result.Collection)
+
+    for (let c of collections) {
+      const getProperty = async () => {
+        let results = await this.#perspective.infer(`subject_class("${this.#subjectClass}", C), collection_getter(C, "${tempId}", "${c}", Value)`)
+        if (results && results.length > 0 && results[0].Value) {
+          return flattenPrologList(eval(results[0].Value))
+        } else {
+          return []
+        }
+      }
+
+      this[c] = await getProperty()
+    }
+
+    this.#baseExpression = tempId;
+
+    return this
+  }
+
+  private async setProperties(key: string, value: any) {
```

Comment:
This only sets one property, right?
```suggestion
  private async setProperty(key: string, value: any) {
```